runtime/ops/io.rs

```diff
@@ -436,6 +429,7 @@ impl StdFileResource {
 
   async fn write(self: Rc<Self>, buf: ZeroCopyBuf) -> Result<usize, AnyError> {
     let mut inner = self.inner.clone();
+    let _permit = self.order_semaphore.acquire().await.unwrap();
```

Comment:
Just verified that this is called while still executing blocking. Also, on the line immediately before the acquire here, the data is in order:

```
...
Hello, world! 20
Hello, world! 21
Hello, world! 22
...
```

But then after the semaphore it's (I'm not sure what `xa` is):

```
...
Hello, world! 20
Hello, Hello world!
Hello, world! 22
...
Hello, world! 99
world! 21xa
```